src/toil/provisioners/aws/awsProvisioner.py

```diff
@@ -1089,17 +1089,21 @@ def _getBoto3BlockDeviceMappings(cls, type_info: InstanceType, rootVolSize: int
         return bdms
 
     @awsRetry
-    def _getNodesInCluster(self, instance_type: Optional[str] = None, preemptable=False, both=False) -> List[Boto2Instance]:
+    def _getNodesInCluster(self, instance_type: Optional[str] = None, preemptable=False, both=False, include_stopped_nodes=False) -> List[Boto2Instance]:
```

Comment:
This is starting to get complicated, especially since "both" means both preemptable and non-preemptable, which is less clear from context now that we also care about whether nodes are stopped or not.

Maybe we should at some point look at when this gets called and what queries are really used, and turn the queries into their own functions?

Reply:
Good point! I will make an appropriate change.